orderer/common/follower/follower_chain.go

```diff
@@ -354,7 +357,14 @@ func (c *Chain) pull() error {
 	// Trigger creation of a new consensus.Chain.
 	c.logger.Info("Block pulling finished successfully, going to switch from follower to a consensus.Chain")
 	c.halt()
-	c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID())
+	for {
+		if isExec := c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID()); isExec ||
+			c.removingExternal.Load() == 1 {
+			break
+		}
+
+		time.Sleep(50 * time.Millisecond)
```

Comment:
Can you please try to refactor it to avoid a busy-wait loop here?

Reply:
I can replace it with such a loop, but will it be more readable and understandable?

```
for ok := c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID()); !ok; ok = c.chainCreator.SwitchFollowerToChain(c.ledgerResources.ChannelID()) {
		if c.removingExternal.Load() == 1 {
			break
		}

		time.Sleep(50 * time.Millisecond)
	}
```

Reply:
done.